src/main.rs

```diff
@@ -1,29 +1,118 @@
-use server::ServerConfig;
+use config::ServerConfig;
+use error::{Error, ErrorKind};
+use node::HttpGrinNode;
+use wallet::HttpWallet;
+
+use clap::App;
+use grin_util::{StopState, ZeroingString};
+use std::env;
+use std::io::{self, Write};
+use std::path::PathBuf;
+use std::sync::Arc;
 
 #[macro_use]
 extern crate lazy_static;
 
+#[macro_use]
+extern crate clap;
+
+mod config;
 mod error;
+mod node;
 mod onion;
 mod secp;
-mod ser;
 mod server;
 mod types;
+mod wallet;
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let secret_key = secp::insecure_rand_secret()?; // todo - load from encrypted key file
-    let server_config = ServerConfig {
-        key: secret_key,
-        addr: "127.0.0.1:3000".parse().unwrap(),
-        is_first: true
-    };
+const DEFAULT_INTERVAL: u32 = 12 * 60 * 60;
 
-    let shutdown_signal = async move {
-        // Wait for the CTRL+C signal
-        tokio::signal::ctrl_c()
-            .await
-            .expect("failed to install CTRL+C signal handler");
+fn main() {
+	if let Err(e) = real_main() {
+        io::stderr().write_all(format!("mwixnet server exited with error:\n{}\n", e).as_bytes()).unwrap();
+        std::process::exit(1);
+    }
+
+	std::process::exit(0);
+}
+
+fn real_main() -> std::result::Result<(), Box<dyn std::error::Error>> {
+	let yml = load_yaml!("../mwixnet.yml");
+	let args = App::from_yaml(yml).get_matches();
+    
+    let config_path = match args.value_of("config_file") {
+        Some(path) => PathBuf::from(path),
+        None => {
+            let mut current_dir = env::current_dir()?;
+            current_dir.push("mwixnet-config.toml");
+            current_dir
+        }
     };
+    let password = args.value_of("pass").ok_or(Error::new(ErrorKind::LoadConfigError))?;
```

Comment:
what do you think about create a variable for each arguments you could can validate each arguments before using them, and instead use the variable directly?